### ISSUE TITLE
Go-Server won't compile (Unused imports + Some types are not generated)

### ISSUE THREAD

**Issue by Hamdy** [Jumpscale/go-raml#288]
**RAML file**
https://github.com/g8os/sdstor/blob/master/specs/raml/sdstor.raml

**COMMAND**
`
$ go-raml server  -l go --dir ~/go/src/github.com/g8os/sdstor/server --ramlfile ~/go/src/github.com/g8os/sdstor/specs/raml/sdstor.raml --import-path github.com/g8os/sdstor/server`

**Errors**
```
server/NamespacesNsidReservationPostRespBody.go:4: imported and not used: "gig_cockpit/raml_definitions/librairies/reservation"
server/NamespacesNsidReservationPostRespBody.go:10: undefined: Reservation
server/namespaces_api.go:80: undefined: ObjectCreate
server/namespaces_api.go:103: undefined: Reservation
server/namespaces_api.go:124: undefined: ReservationRequest
server/namespaces_api.go:152: undefined: ReservationRequest
server/namespaces_api.go:191: undefined: Nsid
server/namespaces_api.go:200: undefined: Nsid

```

**Comment by iwanbk** [comment]
One of the obvious reason is that .raml uses remote .raml file which is not currently supported in master.
WIP can be found in `libraries-in-file` branch

**Comment by iwanbk** [comment]
See #282 for the progress of remote .raml

**Comment by iwanbk** [comment]
where i can find this .raml?
```
../../../../gig-cockpit/raml-definitions/librairies/reservation.raml
```

**Comment by iwanbk** [comment]
> See #282 for the progress of remote .raml

merged to master, please see `--lib-root-urls` command option to use it.

example:

with command line option
```
--lib-root-urls=https://raw.githubusercontent.com/gig-cockpit/raml-definitions/master
```

This line
```
!include https://raw.githubusercontent.com/gig-cockpit/raml-definitions/master/resourceTypes/member.raml
```
The generated code is going to the same as `!include resourceTypes/member.raml`

I'm not sure about this
```
 reservation: ../../../../gig-cockpit/raml-definitions/librairies/reservation.raml
```
in which package/dir the code should be generated?

**Comment by Hamdy** [comment]
> where i can find this .raml?
> reservation: ../../../../gig-cockpit/raml-definitions/librairies/reservation.raml

This is the repo containing this file: [https://github.com/gig-cockpit/raml-definitions](https://github.com/gig-cockpit/raml-definitions)

> in which package/dir the code should be generated?

I'm not sure what do u mean by this question, but the code generated should be included as a part of the server code I'm trying to generate in order for the server to build out of the box

Ps: I still having issues generating the server code using master branch and
`go-raml server  -l go --dir ~/go/src/github.com/g8os/sdstor/server --ramlfile ~/go/src/github.com/g8os/sdstor/specs/raml/sdstor.raml --import-path github.com/g8os/sdstor/server --lib-root-urls=https://raw.githubusercontent.com/gig-cockpit/raml-definitions/master`

**Comment by iwanbk** [comment]
> I'm not sure what do u mean by this question, but the code generated should be included as a part of the server code I'm trying to generate in order for the server to build out of the box

I mean, the .raml file is not in the same root directory as `~/go/src/github.com/g8os/sdstor/specs/raml/sdstor.raml`, so i don't know where to put the generated code.


> Ps: I still having issues generating the server code using master branch

Yeah, it is expected, because you still have this line.
 `reservation: ../../../../gig-cockpit/raml-definitions/librairies/reservation.raml`

It has the problem i mentioned above

**Comment by iwanbk** [comment]
I've tried that raml, indeed still has some error,  more details below.

Changing 
```
reservation: ../../../../gig-cockpit/raml-definitions/librairies/reservation.raml
```
to
```
reservation: https://raw.githubusercontent.com/Jumpscale-Cockpit/raml-definitions/master/librairies/reservation.raml
```

With this command
```
go-raml server --dir ~/pathgo/src/github.com/g8os/sdstor/server --ramlfile sdstor.raml --lib-root-urls=https://raw.githubusercontent.com/Jumpscale-Cockpit/raml-definitions/master
```

error when build
```
# github.com/g8os/sdstor/server
./NamespacesNsidReservationPostRespBody.go:9: undefined: Reservation
./namespaces_if.go:7: imported and not used: "github.com/g8os/sdstor/server/librairies/reservation"
```

**Comment by iwanbk** [comment]
i've checked all the errors:

> ./namespaces_api.go:80: undefined: ObjectCreate

bug in .raml 
`<<resourcePathName | !singularize | !uppercamelcase>>Create` indeed means `ObjectCreate`, while `ObjectCreate` is not defined

> ./namespaces_api.go:191: undefined: Nsid
> ./namespaces_api.go:200: undefined: Nsid

bug in .raml, similar bug like `ObjectCreate` above

> ./NamespacesNsidReservationPostRespBody.go:9: undefined: Reservation
> ./namespaces_api.go:103: undefined: Reservation
> ./namespaces_api.go:124: undefined: ReservationRequest
> ./namespaces_api.go:152: undefined: ReservationRequest

Bug in the go-raml.
The resource type is not properly inherited when it reside in a library

> ./namespaces_if.go:7: imported and not used: "github.com/g8os/sdstor/server/librairies/reservation"

Bug in go-raml, seems like a funny bug :(

**Comment by iwanbk** [comment]
> ./namespaces_if.go:7: imported and not used: "github.com/g8os/sdstor/server/librairies/reservation"
> Bug in go-raml, seems like a funny bug :(

This wrong import is fixed at #294

**Comment by iwanbk** [comment]
This issue, other than .raml spec bugs i mentioned above, should be fixed by #296.
Will merge to master after doing more check.
